src/game_engine/unity/il2cpp.rs

```diff
@@ -672,32 +659,15 @@ impl<const CAP: usize> UnityPointer<CAP> {
             };
 
             cache.offsets[i] = current_offset;
-
-            current_instance_pointer = class_instance + current_offset;
-            cache.current_instance_pointer = Some(current_instance_pointer);
             cache.resolved_offsets += 1;
+
+            current_object =
+                process.read_pointer(current_object + current_offset, module.pointer_size)?;
         }
 
         Ok(())
     }
 
-    /// Dereferences the pointer path, returning the memory address of the value of interest
-    pub fn deref_offsets(
-        &self,
-        process: &Process,
-        module: &Module,
-        image: &Image,
-    ) -> Result<Address, Error> {
-        self.find_offsets(process, module, image)?;
-        let cache = self.cache.borrow();
-        let mut address = cache.base_address;
-        let (&last, path) = cache.offsets[..self.depth].split_last().ok_or(Error {})?;
-        for &offset in path {
-            address = process.read_pointer(address + offset, module.pointer_size)?;
-        }
-        Ok(address + last)
-    }
-
```

Comment:
Why is `deref_offsets` being removed here?

Reply:
I don't really see any use for it, so I wanted to remove unnecessary stuff. You can easily get the address of a certain field in a lot of different ways.

Reply:
My use for it: When I have a `UnityPointer` that points to an `Address` that might be an `Address64` or an `Address32` depending on pointer size, I used `read_pointer` of `deref_offsets` to get that `Address` value without having to match on the possible pointer sizes.

I don't really need `deref_offsets` specifically, but the composition of it with `read_pointer` is something I use a lot.

Reply:
Honestly I have no problem in reintroducing it if you find it useful.